integration/k8s_fixture_test.go

```diff
@@ -168,8 +179,125 @@ func (f *k8sFixture) ClearNamespace() {
 	f.ClearResource("services")
 }
 
+func (f *k8sFixture) getOldKubeConfig() {
+	usr, _ := user.Current()
+	dir := usr.HomeDir
+	// NOTE(dmiller): this assumes that your kube config exists in the default location, and is the one currently being used.
+	// This is true in CI and on my machine, but is not universally true for anyone who might want to run this test on their machine.
+	kubeConfigPath := filepath.Join(dir, ".kube", "config")
+
+	old, err := ioutil.ReadFile(kubeConfigPath)
+	if err != nil {
+		f.t.Fatalf("Error reading KUBECONFIG: %v", err)
+	}
+	info, err := os.Stat(kubeConfigPath)
+	if err != nil {
+		f.t.Fatalf("Error stat'ing KUBECONFIG: %v", err)
+	}
+
+	f.oldKubeConfigPath = kubeConfigPath
+	f.oldKubeConfigPerms = info.Mode()
+	f.oldKubeConfig = string(old)
+}
+
+func (f *k8sFixture) maybeRestoreOldKubeConfig() {
+	if f.oldKubeConfig != "" {
+		err := ioutil.WriteFile(f.oldKubeConfigPath, []byte(f.oldKubeConfig), f.oldKubeConfigPerms)
+		if err != nil {
+			f.t.Fatalf("Error restoring old KUBECONFIG: %v", err)
+		}
+	}
+}
+
+func (f *k8sFixture) createUser() {
+	outWriter := bytes.NewBuffer(nil)
+	cmd := exec.CommandContext(f.ctx, "kubectl", "apply", "-f", "access.yaml")
+	cmd.Stdout = outWriter
+	cmd.Stderr = outWriter
+	cmd.Dir = packageDir
+	err := cmd.Run()
+	if err != nil {
+		f.t.Fatalf("Error creating user: %v. Logs:\n%s", err, outWriter.String())
+	}
+}
+
+func (f *k8sFixture) getSecrets() {
+	outWriter := bytes.NewBuffer(nil)
+	cmdStr := `kubectl get secrets -n tilt-integration -o json | jq -r '.items[] | select(.metadata.name | startswith("tilt-integration-user-token-")) | .data.token'`
```

Comment:
😢

Reply:
diff between these two blocks is not immediately apparent, consider comments for readability